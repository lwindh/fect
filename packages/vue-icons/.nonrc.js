const { join } = require('path')

module.exports = {
  entry: join(__dirname, 'packages'),
  formats: 'noumd',
}
