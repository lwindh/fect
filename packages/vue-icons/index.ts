import { build } from './src/bundler'

//  iife ~
;(() => build())()
