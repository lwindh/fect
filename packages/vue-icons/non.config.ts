import path from 'path'
import { defineNonConfig } from '@fect-ui/cli'

export default defineNonConfig({
  entry: path.join(__dirname, 'packages'),
  formats: 'noumd',
  lib: {
    input: path.join(__dirname, 'packages'),
    format: 'noumd'
  }
})
