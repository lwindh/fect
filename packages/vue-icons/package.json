{
  "name": "@fect-ui/vue-icons",
  "version": "1.0.7",
  "author": "XeryYue",
  "description": "Well prepared high quality Icon Library for fect",
  "scripts": {
    "build": "ts-node index.ts && rm -rf svg && rm -rf packages"
  },
  "main": "lib/index.js",
  "module": "es/index.js",
  "types": "types/index.d.ts",
  "license": "MIT",
  "files": [
    "LISCENSE",
    "es/",
    "lib/",
    "types/",
    "README.md"
  ],
  "devDependencies": {
    "@fect-ui/cli": "1.0.0",
    "ts-node": "^10.3.0",
    "webfont": "^11.2.26",
    "jsdom": "^16.6.0",
    "@types/svgo": "1.3.2",
    "svgo": "1.3.2",
    "@types/jsdom": "^16.2.11",
    "ora": "5.4.1"
  },
  "peerDependencies": {
    "vue": "^3.0.11"
  },
  "homepage": "https://github.com/fay-org/fect",
  "repository": {
    "type": "git",
    "url": "https://github.com/fay-org/fect"
  }
}
