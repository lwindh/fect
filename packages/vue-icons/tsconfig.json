{
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "./lib",
    "target": "ES2015",
    "module": "CommonJS",
    "sourceMap": false,
    "declaration": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "importHelpers": true,
    "allowJs": true,
    "jsx": "preserve",
    "lib": ["esnext", "dom"]
  },
  "include": ["src/**/*"]
}
