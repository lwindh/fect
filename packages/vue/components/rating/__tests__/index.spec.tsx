import { mount } from '@vue/test-utils'
import Rating from '../index'
import { github } from '@fect-ui/vue-icons'

describe('Rating', () => {
  it('should be render as a element', () => {
    const wrapper = mount(Rating)
    expect(() => wrapper.unmount).not.toThrow()
  })

  it('should be support custom Icon render', () => {
    const custom = () => <github />
    const wrapper = mount(Rating, {
      props: {
        icon: custom(),
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be support different types', () => {
    const _success = mount(Rating, {
      props: {
        type: 'success',
      },
    })

    const _warn = mount(Rating, {
      props: {
        type: 'warning',
      },
    })

    const _error = mount(Rating, {
      props: {
        type: 'error',
      },
    })

    const _other = mount(Rating, {
      props: {
        type: undefined,
      },
    })

    expect(_success.html()).toMatchSnapshot()
    expect(_warn.html()).toMatchSnapshot()
    expect(_error.html()).toMatchSnapshot()
    expect(_other.html()).toMatchSnapshot()
  })

  it('should be support custom count & lock', () => {
    const wrapper = mount(Rating, {
      props: {
        count: 3,
        locked: true,
      },
    })
    expect(wrapper.findAll('.fect-rating__box').length).toBe(3)
    expect(wrapper.findAll('.is-locked')).toBeTruthy()
  })

  it('should be acitved', async () => {
    const wrapper = mount(Rating, {
      props: {
        modelValue: 3,
      },
    })
    expect(wrapper.findAll('.hoverd').length).toEqual(3)
  })

  it('should be support change Event', async () => {
    const wrapper = mount(Rating, {
      props: {
        modelValue: 0,
      },
    })

    const els = wrapper.findAll('.fect-rating__box')
    await els[1].trigger('click')
    expect(wrapper.emitted('update:modelValue')![0]).toEqual([2])
    await wrapper.setProps({ modelValue: 2 })
    expect(wrapper.emitted('change')![0]).toEqual([2])
  })

  it('when trigger mouseenvt , should be actived', async () => {
    const wrapper = mount(Rating, {
      props: {
        modelValue: 0,
      },
    })

    const els = wrapper.findAll('.fect-rating__box')
    await els[1].trigger('mouseenter')
    expect(wrapper.find('.hoverd').exists()).toBe(true)
    await els[1].trigger('mouseleave')
    expect(wrapper.find('.hoverd').exists()).toBe(false)
  })

  it('should disabled all event when set locked', async () => {
    const wrapper = mount(Rating, {
      props: {
        modelValue: 0,
        locked: true,
      },
    })
    const els = wrapper.findAll('.fect-rating__box')
    await els[1].trigger('click')
    expect(wrapper.find('.hoverd').exists()).toBe(false)
    await els[1].trigger('mouseenter')
    expect(wrapper.find('.hoverd').exists()).toBe(false)
    await els[1].trigger('mouseleave')
    expect(wrapper.find('.hoverd').exists()).toBe(false)
  })
})
