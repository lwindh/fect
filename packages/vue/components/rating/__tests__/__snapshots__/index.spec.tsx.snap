// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should be support custom Icon render 1`] = `
<div class="fect-rating" role="slider" tabindex="0">
  <div class="fect-rating__box  " style="--rating-svg-color: var(--primary-foreground);"><svg fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" shape-rendering="geometricPrecision" viewBox="0 0 24 24">
      <path d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 00-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0020 4.77 5.07 5.07 0 0019.91 1S18.73.65 16 2.48a13.38 13.38 0 00-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 005 4.77a5.44 5.44 0 00-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 009 18.13V22"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--primary-foreground);"><svg fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" shape-rendering="geometricPrecision" viewBox="0 0 24 24">
      <path d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 00-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0020 4.77 5.07 5.07 0 0019.91 1S18.73.65 16 2.48a13.38 13.38 0 00-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 005 4.77a5.44 5.44 0 00-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 009 18.13V22"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--primary-foreground);"><svg fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" shape-rendering="geometricPrecision" viewBox="0 0 24 24">
      <path d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 00-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0020 4.77 5.07 5.07 0 0019.91 1S18.73.65 16 2.48a13.38 13.38 0 00-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 005 4.77a5.44 5.44 0 00-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 009 18.13V22"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--primary-foreground);"><svg fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" shape-rendering="geometricPrecision" viewBox="0 0 24 24">
      <path d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 00-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0020 4.77 5.07 5.07 0 0019.91 1S18.73.65 16 2.48a13.38 13.38 0 00-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 005 4.77a5.44 5.44 0 00-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 009 18.13V22"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--primary-foreground);"><svg fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" shape-rendering="geometricPrecision" viewBox="0 0 24 24">
      <path d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 00-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0020 4.77 5.07 5.07 0 0019.91 1S18.73.65 16 2.48a13.38 13.38 0 00-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 005 4.77a5.44 5.44 0 00-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 009 18.13V22"></path>
    </svg></div>
</div>
`;

exports[`Rating should be support different types 1`] = `
<div class="fect-rating" role="slider" tabindex="0">
  <div class="fect-rating__box  " style="--rating-svg-color: var(--success-default);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--success-default);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--success-default);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--success-default);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--success-default);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
</div>
`;

exports[`Rating should be support different types 2`] = `
<div class="fect-rating" role="slider" tabindex="0">
  <div class="fect-rating__box  " style="--rating-svg-color: var(--warning-default);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--warning-default);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--warning-default);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--warning-default);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--warning-default);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
</div>
`;

exports[`Rating should be support different types 3`] = `
<div class="fect-rating" role="slider" tabindex="0">
  <div class="fect-rating__box  " style="--rating-svg-color: var(--error-default);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--error-default);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--error-default);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--error-default);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--error-default);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
</div>
`;

exports[`Rating should be support different types 4`] = `
<div class="fect-rating" role="slider" tabindex="0">
  <div class="fect-rating__box  " style="--rating-svg-color: var(--primary-foreground);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--primary-foreground);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--primary-foreground);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--primary-foreground);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
  <div class="fect-rating__box  " style="--rating-svg-color: var(--primary-foreground);"><svg viewBox="0 0 24 24" width="24" height="24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
      <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"></path>
    </svg></div>
</div>
`;
